### PR TITLE
Dm devguide

### PR DESCRIPTION
**Pull request by Yubi-David**
Updated with updates not saved and pushed in the last update

### CODE REVIEW

File: content/Developer_Program/Guides/Securing_Web_Resources.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+= Securing Web Sites, Services and Applications
+The internet is quickly evolving from its roots as a passive channel for displaying requested information to an interactive platform replacing traditional desktop OS’s for applications and services. However, a decentralized, cloud-based, universally accessible platform has its unique challenges to securing accounts and services. The YubiKey offers multiple solutions for adding multifactor authentication compatible with services of any type.
+
+When choosing an option, it is important to understand all of the options available.
+
+== FIDO2 / WebAuthn
+The link:https://fidoalliance.org/[FIDO Alliance] has curated the new trails in blazed by innovators in secure authentication, providing a standard, open platform for maintaining new specifications. WebAuthn represents a new way of performing authentication using public key cryptography, but without the overhead of a Public Key Infrastructure framework or a central trust authority.
```
Comment (FFincannon) [inline content/Developer_Program/Guides/Securing_Web_Resources.adoc]: The first "in" is superfluous

File: content/Developer_Program/Guides/Touch_triggered_OTP.adoc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+==== Yubico OTP Challenge-Response
+The Challenge-Response mode may also be used to generate a Yubico OTP with a 6 byte string (similar to the private ID) passed to the YubiKey in the challenge. The Yubico OTP mode takes a 6 byte challenge and creates a response using the Yubico OTP algorithm and a user defined AES key, where variable fields generated by the device creates different responses even if the challenge is the same. When
+
+== Introduction to Static Password
+For complete legacy support, the YubiKey Touch-Triggered OTP Slots can also hold a static password. A static password is an unchanging string of characters which remain the same each time the OTP slot is triggered, passed as a series of keystrokes, exactly like a password users would enter directly. However, the YubiKey offers the advantage that the password is entered the same every time, and even if the YubiKey hardware is left in plain sight, the password is not exposed to the casual glance or recording.
```
[the remark this answers is not in the document]
Comment (Yubi-David) [reply content/Developer_Program/Guides/Touch_triggered_OTP.adoc]: Video recording - added for clarity.

File: content/Developer_Program/Guides/Touch_triggered_OTP.adoc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+* link:https://developers.yubico.com/OTP/Specifications/OTP_validation_protocol.html[OTP Validation Protocol]
+* link:https://developers.yubico.com/OTP/Specifications/Test_vectors.html[Test vectors]
+
+Servers:
+
+* Validation Server: link:https://developers.yubico.com/yubikey-val/[yk-val]
+* KSM server (both using YubiHSM and soft DB): link:https://developers.yubico.com/python-pyhsm/YubiKey_KSM.html[pyHSM]
+
+==== YubiCloud
+For services and websites connected to the internet, Yubico offers a free Yubico OTP Validation service called the YubiCloud. Every YubiKey is programmed at the factory with a YubiCloud credential, removing the need to manage and upload secrets. The YubiCloud runs the same code and behaves in the same manner as a Yubico OTP Validation servers available as open source. For more details, refer to the link:https://support.yubico.com/support/solutions/articles/15000006482-yubicloud-otp-validation-service-guide[YubiCloud OTP Validation Service Guide].
```
[the remark this answers is not in the document]
Comment (Yubi-David) [reply content/Developer_Program/Guides/Touch_triggered_OTP.adoc]: Removed reference and link to old architecture doc.

File: content/Developer_Program/Guides/Touch_triggered_OTP.adoc [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+=== OATH-HOTP Validation
+The HOTP code is created by hashing the secret key with the counter value, and truncating the end result to the desired length of the OTP code. This code is then sent to the authentication service, where it is compared against the results of the same calculation done by the server against its internal counter.
+
+Since the OTP validation is done by a comparison check, no data on the counter value of the OATH-HOTP OTP counter on the YubiKey can be extracted. This means if the counter value on the YubiKey and the authentication server fall out of synchronization, such as if the YubiKey generates a number of OATH-HOTP codes without submitting them to the server. To reduce the chance of an out-of-sync event, most OATH-HOTP Authentication servers have a “look-ahead” window, checking the OTPs generated with the n number of counter values. However, should the counter value on the authenticator exceed this window, the authentication server will still fail. Many OATH-HOTP services have a recovery path in the event this occurs - the details on the process is specific to each service.
```
[the remark this answers is not in the document]
Comment (Yubi-David) [reply content/Developer_Program/Guides/Touch_triggered_OTP.adoc]: Updated to remove "if"

**Pull request merged**